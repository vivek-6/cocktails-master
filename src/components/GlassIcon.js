import React from "react";
import SvgIcon from "@material-ui/core/SvgIcon";
import { ReactComponent as Martini } from "../images/glasses/Martini.svg";

// Glass SVGs
// Note that currently not all of these are completed, these are just placeholders
// for if someone at some point makes some real SVGs. Note that there are
// a few constraints on the SVG format for material UI's SVG icon;
//
// https://material-ui.com/style/icons/#svg-icons
//
// * SVG elements should be scaled for a 24x24px viewport.
// * The first child should be a 'path' element.
//
import { ReactComponent as OldFashioned } from "../images/glasses/OldFashioned.svg";
import { ReactComponent as Collins } from "../images/glasses/Collins.svg";
import { ReactComponent as Highball } from "../images/glasses/Highball.svg";
import { ReactComponent as ChampagneFlute } from "../images/glasses/ChampagneFlute.svg";
import { ReactComponent as ChampagneTulip } from "../images/glasses/ChampagneTulip.svg";
import { ReactComponent as Margarita } from "../images/glasses/Margarita.svg";
import { ReactComponent as Hurricane } from "../images/glasses/Hurricane.svg";
import { ReactComponent as Shot } from "../images/glasses/Shot.svg";
import { ReactComponent as HotDrink } from "../images/glasses/HotDrink.svg";
import { ReactComponent as WhiteWine } from "../images/glasses/WhiteWine.svg";

const glassSvgLookup = {
  martini: Martini,
  "old-fashioned": OldFashioned,
  collins: Collins,
  highball: Highball,
  "champagne-flute": ChampagneFlute,
  margarita: Margarita,
  "champagne-tulip": ChampagneTulip,
  hurricane: Hurricane,
  shot: Shot,
  "hot-drink": HotDrink,
  "white-wine": WhiteWine
};

const GlassIcon = ({ glass = "martini", ...otherProps }) => {
  const GlassSvg = glassSvgLookup[glass.toString()];

  return (
    <SvgIcon {...otherProps}>
      <GlassSvg />
    </SvgIcon>
  );
};

export default GlassIcon;
