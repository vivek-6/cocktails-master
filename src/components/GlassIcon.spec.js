import GlassIcon from "./GlassIcon";
import React from "react";
import renderer from "react-test-renderer";

const glasses = [
  "martini",
  "old-fashioned",
  "collins",
  "highball",
  "champagne-flute",
  "margarita",
  "champagne-tulip",
  "hurricane",
  "shot",
  "hot-drink",
  "white-wine"
];

it("does not explode when rendered", async () => {
  const tree = renderer.create(
    <>
      {glasses.map(g => (
        <GlassIcon key={g} glass={g} />
      ))}
    </>
  );

  expect(tree).toMatchSnapshot();
});
