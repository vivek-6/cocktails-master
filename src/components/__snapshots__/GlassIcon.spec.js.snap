// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not explode when rendered 1`] = `
Array [
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      Martini.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      OldFashioned.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      Collins.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      Highball.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      ChampagneFlute.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      Margarita.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      ChampagneTulip.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      Hurricane.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      Shot.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      HotDrink.svg
    </svg>
  </svg>,
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg>
      WhiteWine.svg
    </svg>
  </svg>,
]
`;
